core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -573,7 +574,23 @@ func (ec *EthConfirmer) batchFetchReceipts(ctx context.Context, attempts []EthTx
 		}
 
 		if receipt.Status == 0 {
-			l.Warnf("transaction %s reverted on-chain", receipt.TxHash)
+			_, err := ec.ethClient.CallContract(context.Background(), ethereum.CallMsg{
+				From:       attempt.EthTx.FromAddress,
+				To:         &attempt.EthTx.ToAddress,
+				Gas:        uint64(attempt.EthTx.GasLimit),
+				GasPrice:   attempt.GasPrice.ToInt(),
+				GasFeeCap:  attempt.GasFeeCap.ToInt(),
+				GasTipCap:  attempt.GasTipCap.ToInt(),
+				Value:      nil,
+				Data:       attempt.EthTx.EncodedPayload,
+				AccessList: nil,
+			}, receipt.BlockNumber)
+			revReason, err := evmclient.ExtractRevertReasonFromRPCError(err)
+			if err == nil {
+				l.Warnf("transaction %s reverted on-chain revert reason %s", receipt.TxHash, revReason)
+			} else {
+				l.Warnf("transaction %s reverted on-chain unable to extract revert reason %v", receipt.TxHash, err)
```

Comment:
Would the `Warnw()` variant fit better? :shrug:
```suggestion
				l.Warnw("transaction %s reverted on-chain", receipt.TxHash, "reason", revReason)
			} else {
				l.Warnw("transaction %s reverted on-chain: unable to extract revert reason", receipt.TxHash, "err", err)
```
Should we have special handling for the error when querying too late on a non-archive node? Or will that be clear already? ('state trie missing' or something iirc?)

Reply:
that would be rare given we do this upon receipt fetching, but yeah just state trie missing which I think is ok